Application/Items/Commands/DeleteItemCommand.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Application.Common.Exceptions;
+using Application.Common.Interfaces;
+using Domain.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Items;
+
+public record DeleteItemCommand(int Id) : IAuthorizedRequest
+{
+    public async Task<bool> Authorize(Employee employee, IUserService userService, IApplicationDbContext dbContext)
+    {
+        return await userService.CanManageItemAsync(employee, Id);
```

Comment:
canManageTenant

Reply:
this is how the method would look like then:
```suggestion
        var item = await dbContext.Items.FindAsync(Id);
        return item != null ? await userService.CanManageTenantAsync(employee, item.TenantId) : true;
```
wouldn't it be nicer to have a separate `userService` method `CanManageItemAsync()` then?

Reply:
why im against it: user service would be filled up with functions which would be used only in few places and few times, whereas canviewtenant and canmanagetenant can be used in every command. f.e. you'll use the canviewitem only in Items queries or commands

personally, I'd write the logic in the authorize method instead

Reply:
ok then, I previously thought that logic in Authorize method would be bad, but if it's fine, i'll rewrite it there